terraform/eval_variable.go

```diff
@@ -81,6 +77,12 @@ func (n *EvalTypeCheckVariable) Eval(ctx EvalContext) (interface{}, error) {
 			case map[string]interface{}:
 				continue
 			default:
+				if list, ok := proposedValue.([]interface{}); ok && len(list) == 1 {
+					if m, ok := list[0].(map[string]interface{}); ok {
+						n.Variables[name] = m
+						continue
+					}
+				}
```

Comment:
I don't like doing the coercion mid-type-check. I'm planning to pull this out into its own `EvalNode`.

Just wanted to get visibility on the strategy before i go further